packages/service-library/src/servicelib/redis.py

```diff
@@ -61,35 +88,79 @@ async def ping(self) -> bool:
     async def lock_context(
         self,
         lock_key: str,
-        lock_value: Optional[Union[bytes, str]] = None,
+        lock_value: bytes | str | None = None,
+        *,
+        blocking: bool = False,
+        blocking_timeout_s: NonNegativeFloat = 5,
     ) -> AsyncIterator[Lock]:
-        ttl_lock = None
-        try:
+        """Tries to acquire a lock.
+
+        :param lock_key: unique name of the lock
+        :param lock_value: content of the lock, defaults to None
+        :param blocking: should block here while acquiring the lock, defaults to False
+        :param blocking_timeout_s: time to wait while acquire a lock before giving up, defaults to 5
+
+        :raises CouldNotAcquireLockError: reasons why lock acquisition fails:
+            1. `blocking==False` the lock was already acquired by some other entity
+            2. `blocking==True` timeouts out while waiting for lock to be free (another entity holds the lock)
+        """
+
+        total_lock_duration: datetime.timedelta = _DEFAULT_LOCK_TTL
+        lock_unique_id = f"lock_extender_{uuid4()}"
```

Comment:
Suggestion: before was {lock_key}_extender. While I agree that lock_key might not be unique I do think that a uuid only is very unreadable. care to use both?

Reply:
will use both then